concourse/pipelines/deploy-cloudfoundry.yml

```diff
@@ -108,10 +109,50 @@ jobs:
         - get: pipeline-trigger
           passed: [ 'init' ]
           trigger: true
+
+      - task: generate-cf-secrets
+        config:
+          image: docker:///governmentpaas/mksecrets
+          inputs:
+            - name: paas-cf
+          run:
+            path: sh
+            args:
+              - -c
+              - -e
+              - |
+                ./paas-cf/manifests/cf-manifest/scripts/generate-cf-secrets.sh > cf-secrets.yml
+                ls -l cf-secrets.yml
+
+      - task: upload
+        config:
+          image: docker:///governmentpaas/curl-ssl
+          inputs:
+            - name: paas-cf
+            - name: generate-cf-secrets
+          run:
+            path: paas-cf/concourse/scripts/s3init.sh
+            args:
+              - {{state_bucket}}
+              - cf-secrets.yml
+              - generate-cf-secrets/cf-secrets.yml
```

Comment:
This will be an issue when upgrading a deployment with existing secrets in S3 because the new secrets won't get written back to S3. The only way round this at the moment is to manually delete the secrets.yml from S3.